src/synapse/azext_synapse/custom.py

```diff
@@ -121,59 +123,126 @@ def create_workspace(cmd, client, resource_group_name, workspace_name, account_u
+        if max_node_count:
+            existing_spark_pool.auto_scale.min_node_count = max_node_count
+    else:
+        existing_spark_pool.auto_scale = AutoScaleProperties(enabled=enable_auto_scale, min_node_count=min_node_count,
+                                                             max_node_count=max_node_count)
+
+    if existing_spark_pool.auto_pause is not None:
+        if enable_auto_pause is not None:
+            existing_spark_pool.auto_pause.enabled = enable_auto_pause
+        if delay_in_minutes:
+            existing_spark_pool.auto_pause.delay_in_minutes = delay_in_minutes
+    else:
+        existing_spark_pool.auto_pause = AutoPauseProperties(enabled=enable_auto_pause,
+                                                             delay_in_minutes=delay_in_minutes)
+
+    return sdk_no_wait(no_wait, client.create_or_update, resource_group_name, workspace_name, spark_pool_name,
+                       existing_spark_pool, force=force)
 
 
 def delete_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name, no_wait=False):
     return sdk_no_wait(no_wait, client.delete, resource_group_name, workspace_name, spark_pool_name)
 
 
-def create_sql_pool(cmd, client, resource_group_name, workspace_name, sql_pool_name, location, sku_name,
-                    sku_tier=None, max_size_bytes=None, source_database_id=None, recoverable_database_id=None,
-                    create_mode=None, tags=None, no_wait=False):
+def create_sql_pool(cmd, client, resource_group_name, workspace_name, sql_pool_name, sku_name, tags=None,
+                    no_wait=False):
+    # get the location of the workspace
+    workspace_client = cf_synapse_client_workspace_factory(cmd.cli_ctx)
+    workspace_object = workspace_client.get(resource_group_name, workspace_name)
+    location = workspace_object.location
+
+    max_size_bytes = None
+    sku_tier = None
     sku = Sku(tier=sku_tier, name=sku_name)
-    sql_pool_info = SqlPool(sku=sku, location=location, max_size_bytes=max_size_bytes,
-                            source_database_id=source_database_id, recoverable_database_id=recoverable_database_id,
-                            create_mode=create_mode, tags=tags)
+
+    create_mode = 'Default'
+
+    sql_pool_info = SqlPool(sku=sku, location=location, max_size_bytes=max_size_bytes, create_mode=create_mode,
+                            tags=tags)
 
     return sdk_no_wait(no_wait, client.create, resource_group_name, workspace_name, sql_pool_name, sql_pool_info)
+
+
+def update_sql_pool(cmd, client, resource_group_name, workspace_name, sql_pool_name, sku_name=None, tags=None):
+    sku = Sku(name=sku_name)
+    sql_pool_patch_info = SqlPoolPatchInfo(sku=sku, tags=tags)
+    return client.update(resource_group_name, workspace_name, sql_pool_name, sql_pool_patch_info)
+
+
+def create_firewall_rule(cmd, client, resource_group_name, workspace_name, rule_name, start_ip_address, end_ip_address,

```

Comment:
>create_firewall_rule [](start = 4, length = 20)

Just curious. Why do we need this custom method rather than calling SDK method directly?

Reply:
In sdk, start_ip_address  and end_ip_address have default value None.  I think that we had better make them required.

---
In reply to: [417129096](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r417129096) [](ancestors = 417129096)

Reply:
I see. Maybe we should update Swagger spec to remove default value in future?

Reply:
If we add `required=True` then we can remove the custom method. 

---
In reply to: [418063786](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r418063786) [](ancestors = 418063786)